### ISSUE TITLE
Enable BIND_NOW (-Wl,-z,now) when compiling Linux builds 

### ISSUE THREAD

**Issue by biergaizi** [electron/electron#15149]
We've hardened Linux builds by enabling PIE and RELRO in our GN config (#14961), and should continue to try hardening Linux builds by enabling [BIND_NOW](https://wiki.debian.org/Hardening#DEB_BUILD_HARDENING_BINDNOW_.28ld_-z_now.29). These flags are off for all released versions (3.0 and prior). This isn't an active security risk, but it increases attack surface and might be straightforward to add, and help to improve the security of a whole range of applications built upon Electron, including Signal-Desktop.

The Debian wiki mentioned overhead but it should not be an issue, and it has been enabled in official Chrome/Chromium builds since more than five years ago. https://github.com/chromium/chromium/blame/f28a94764fd9d475add499cfb1038110de0a3acc/build/config/compiler/BUILD.gn#L176

Results can be validated with hardening-check.

**Comment by biergaizi** [comment]
BIND_NOW, same, appears to be enabled on the GN build already (inherited somewhere)...

```
$ hardening-check electron 
electron:
 Position Independent Executable: yes
 Stack protected: yes
 Fortify Source functions: no, only unprotected functions found!
 Read-only relocations: yes
 Immediate binding: yes
```
Only need to enable it for 3-0-x and 2-0-x branches.

**Comment by MarshallOfSound** [comment]
Fixed in 2.0, 3.0 and 4.0
